web/src/compose_actions.js

```diff
@@ -187,9 +187,11 @@ export function start(msg_type, opts) {
         return;
     }
 
-    // We may be able to clear it to change the recipient, so save any
-    // existing content as a draft.
-    drafts.update_draft();
+    // Save any existing content as a draft if we go to a different narrow or there
+    // is new message content to populate the compose box with (like from a draft).
+    if (!same_recipient_as_before(msg_type, opts) || opts.content !== undefined) {
+        drafts.update_draft();
+    }
```

Comment:
I'm a little worried about this because this isn't the exact same check as the one for `clear_box`; we always want to call `update_draft` if we're about to destroy content. I could imagine extracting a `const will_clear_content = ...` variable to manage that issue. 

It's really really important that we not make it possible to accidentally lose work, so I think it's worth that kind of intentional construction to make sure that this logic agrees with the logic that will actually clear the compose textarea.

Reply:
@timabbott It isn't the exact same check, yes, but it _is_ a superset of the `clear_box` check. So we will definitely `update_draft` anytime we `clear_box`. 

We don't `clear_box` when we restore a draft in the same narrow, but we do want to draft the current message then too, hence the `|| opts.content` check here. I could've instead used the `draft_id` check, but I think this is more future proof for any situation where we might have content (not from a draft) to populate the compose box on start.

Reply:
Fixed a bug; moved this check a little below, right after all the tweaks to the opts and just before the `clear_box` check, so that `same_recipient_as_before(msg_type, opts)` is the same for both checks.